src/EditorFeatures/CSharpTest/ExtractClass/ExtractClassTests.cs

```diff
@@ -0,0 +1,1443 @@
+{
+    private event EventHandler Event1;
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(input, expected);
+        }
+
+        [Fact]
+        public async Task TestProperty()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test
+{
+    int [||]MyProperty { get; set; }
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test : MyBase
+{
+}
+        </Document>
+        <Document FilePath=""MyBase.cs"">internal class MyBase
+{
+    int MyProperty { get; set; }
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(input, expected);
+        }
+
+        [Fact]
+        public async Task TestField()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test
+{
+    int [||]MyField;
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test : MyBase
+{
+}
+        </Document>
+        <Document FilePath=""MyBase.cs"">internal class MyBase
+{
+    int MyField;
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(input, expected);
+        }
+
+        [Fact]
+        public async Task TestFileHeader()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">// this is my document header
+// that should be copied over
+
+class Test
+{
+    int [||]Method()
+    {
+        return 1 + 1;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">// this is my document header
+// that should be copied over
+
+class Test : MyBase
+{
+}
+        </Document>
+        <Document FilePath=""MyBase.cs"">// this is my document header
+// that should be copied over
+
+internal class MyBase
+{
+    int Method()
+    {
+        return 1 + 1;
+    }
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(input, expected);
+        }
+
+        [Fact]
+        public async Task TestWithInterface()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+interface ITest 
+{
+    int Method();
+}
+
+class Test : ITest
+{
+    int [||]Method()
+    {
+        return 1 + 1;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+interface ITest 
+{
+    int Method();
+}
+
+class Test : MyBase, ITest
+{
+}
+        </Document>
+        <Document FilePath=""MyBase.cs"">internal class MyBase
```

Comment:
do we preseve accessibility based on the starting type? i.e. if the starting type is public, is the new base class public?

Reply:
We do. Added test.